rotkehlchen/tests/exchanges/test_kucoin.py

```diff
@@ -0,0 +1,727 @@
+            f'Kucoin get request at {request_url} connection error: {str(e)}.',
+        ) from e
+
+    if response.status_code != HTTPStatus.OK:
+        raise RemoteError(
+            f'Kucoin query responded with error status code: {response.status_code} '
+            f'and text: {response.text}',
+        )
+    try:
+        response_dict = rlk_jsonloads_dict(response.text)
+    except JSONDecodeError as e:
+        raise RemoteError(f'Kucoin returned invalid JSON response: {response.text}') from e
+
+    # Extract the unique symbols from the exchange pairs
+    unsupported_assets = set(UNSUPPORTED_KUCOIN_ASSETS)
+    for entry in response_dict['data']:
+        symbol = entry['currency']
+        try:
+            asset_from_kucoin(symbol)
+        except UnsupportedAsset:
+            assert symbol in unsupported_assets
+        except UnknownAsset as e:
+            test_warnings.warn(UserWarning(
+                f'Found unknown asset {e.asset_name} in kucoin. '
+                f'Support for it has to be added',
+            ))
+
+
+def test_api_query_retries_request(mock_kucoin):
+
+    def get_response():
+        results = [
+            """{"code":400007,"msg":"unknown error"}""",
+            """{"code":400007,"msg":"unknown error"}""",
+        ]
+        for result_ in results:
+            yield result_
+
+    def mock_api_query_response(url):  # pylint: disable=unused-argument
+        return MockResponse(HTTPStatus.TOO_MANY_REQUESTS, next(get_response))
+
+    get_response = get_response()
+    api_request_retry_times_patch = patch(
+        target='rotkehlchen.exchanges.kucoin.API_REQUEST_RETRY_TIMES',
+        new=1,
+    )
+    api_request_retry_after_seconds_patch = patch(
+        target='rotkehlchen.exchanges.kucoin.API_REQUEST_RETRIES_AFTER_SECONDS',
+        new=0,
+    )
+    api_query_patch = patch.object(
+        target=mock_kucoin.session,
+        attribute='get',
+        side_effect=mock_api_query_response,
+    )
+    with ExitStack() as stack:
+        stack.enter_context(api_request_retry_times_patch)
+        stack.enter_context(api_request_retry_after_seconds_patch)
+        stack.enter_context(api_query_patch)
+        result = mock_kucoin._api_query(
+            options={
+                'currentPage': 1,
+                'pageSize': 500,
+                'tradeType': 'TRADE',
+            },
+            case=KucoinCase.TRADES,
+        )
+
+    assert result.status_code == HTTPStatus.TOO_MANY_REQUESTS
+    errors = mock_kucoin.msg_aggregator.consume_errors()
+    assert len(errors) == 1
+    expected_error = (
+        'Got remote error while querying kucoin trades: Kucoin trades request '
+        'failed after retrying 1 times.'
+    )
+    assert errors[0] == expected_error
+
+
+@pytest.mark.parametrize('should_mock_current_price_queries', [True])
+def test_deserialize_accounts_balances(mock_kucoin, inquirer):  # pylint: disable=unused-argument
+    accounts_data = [
+        {
```

Comment:
these data are taken from your sandbox responses?

Reply:
Yes, this entirely dataset comes from our sandbox account but adding 2 extras: 1 account data with an unexisting symbol (testing the asset deserialization for an unknown token), and one with BCHSV (testing we are able to convert symbols). The thing is that the sandbox is restricted to BTC, ETH, USDT and KCS.